### DISCUSSION TITLE
dev ui available on local kubernetes deployment?

### DISCUSSION THREAD

**Discussion by rgreig** [quarkusio/quarkus#22565] [Q&A] [answered]
I am using Quarkus successfully deploying to a local k8s cluster for development, and the live code reloading all works fine (and is great). However, I would also like to be able to use the dev ui but it doesn't seem to be available and I can't find any configuration options that can be set to make it available. Am I just missing something?

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
> I am using Quarkus successfully deploying to a local k8s cluster for development, and the live code reloading all works fine (and is great). However, I would also like to be able to use the dev ui but it doesn't seem to be available and I can't find any configuration options that can be set to make it available. Am I just missing something?

Are you using remote dev mode?

**Reply by rgreig** [reply]
Yes, I am and the live reloading etc works when I make changes to the code. However, in the k8s cluster I can't access /q/dev (but I can hit /q/swagger-ui etc).

**Reply by geoand** [reply]
Right. So indeed in the remote dev mode we do not enable it.

**Reply by rgreig** [reply]
Was that a deliberate decision or a side effect of the current implementation? Some of the features like the bean explorer would be useful I think.

**Reply by geoand** [reply]
It was deliberate.

Indeed some of the features could work, but a lot would not.

**Reply by rgreig** [reply]
Fair enough. Maybe worth considering for future having some way that dev ui extensions could mark themselves as remote compatible so that those features that are useful in remote mode could still be leveraged?

**Reply by geoand** [reply]
Indeed, that could be useful

**Reply by joes** [reply]
Has anyone considered the possibility of a Quarkus "dev-image" that could be attached to a running pod as an [ephemeral container](https://kubernetes.io/docs/concepts/workloads/pods/ephemeral-containers/) @geoand ?

**Reply by geoand** [reply]
We are being very conservative with anything not related to dev mode. That said, we could of course consider such things

**Reply by maxandersen** [reply]
FYI the reason it's turned of for remote dev is that you then most likely have a endpoint where you are able to browse security keys. Ie. A security risk.

**Reply by joes** [reply]
Would it be possible/acceptable to expose q/dev on an alternative port that is not made available to the world? This port could be exposed as a ClusterIP-service and to access the service the developer could use `kubectl port-forward` on their machine?

**Reply by lukens** [reply]
I'm new to Quarkus and was assuming this was something broken in my setup. Could the Dev UI in remote mode perhaps instead display a message like "Dev UI not available in remote-dev mode", or similar?

Also, for me (like I suspect a lot of other people using remote-dev) the "remote" development environment is actually a local Kubernetes cluster, and so the fears about opening sensitive data to the world don't apply so much, so maybe there could be a configuration parameter to enable the Dev UI?

**Reply by geoand** [reply]
> I'm new to Quarkus and was assuming this was something broken in my setup. Could the Dev UI in remote mode perhaps instead display a message like "Dev UI not available in remote-dev mode", or similar?

I think this makes sense
